src/animation_loader.rs

```diff
@@ -1,30 +1,46 @@
-use bevy::prelude::*;
+use bevy::{
+    asset::{Asset, Handle},
+    ecs::system::SystemParamItem,
+    prelude::*,
+    render::render_asset::{self, RenderAsset, RenderAssetPlugin, RenderAssetUsages},
+};
 use bvh_anim::{errors::LoadError, Bvh};
 use std::{fs, io::BufReader};
 
 pub struct AnimationLoaderPlugin;
 
 impl Plugin for AnimationLoaderPlugin {
     fn build(&self, app: &mut App) {
-        app.add_systems(Startup, store_bvh);
-        app.insert_resource(BvhData::new());
+        app.add_plugins(RenderAssetPlugin::<BvhAsset>::default())
+            .init_asset::<BvhAsset>()
+            .add_systems(Startup, store_bvh);
     }
 }
 
-#[derive(Resource)]
-pub struct BvhData {
-    pub bvh_animation: Vec<Bvh>,
+#[derive(Component, Clone, Default)]
+pub struct BvhHandle {
+    pub bvh_assets: Handle<BvhAsset>,
 }
```

Comment:
I would suggest storing all of these data in a `Vec` as a `Resource` instead.
```suggestion
#[derive(Resource, Default, Clone)]
pub struct BvhHandles {
    pub handles: Vec<Handle<BvhAsset>>,
}
```

Reply:
FYI, a `Handle<T>` by itself is already a component (see [source](https://docs.rs/bevy_asset/0.13.2/src/bevy_asset/handle.rs.html#125))